### DISCUSSION TITLE
With the new Client Extension, how to build Multi Tenant prisma client?

### DISCUSSION THREAD

**Discussion by masterbater** [prisma/prisma#19917] [Q&A] [answered]
Hello guys, can you share how you create your own client extension to append tenantId to all models
```typescript
const xprisma = new prisma.$extends({
  model: {
    $allModels: {
      validate(data: unknown) {
        /* Generic validation logic */
      },
    },
  },
})
```
I want to validate first if it exist,then proceed to add tenant id to all queries

**Comment by nurul3101** [comment]
Hello @masterbater 👋 

Do you mean you want to include a field called `tenantId` to all the models defined in `schema.prisma` file?

**Reply by masterbater** [reply]
With this trpc code I want to create my own prisma client, base on session there is a tenantId, 


```typescript
export const createInnerTRPCContext = async (opts: CreateContextOptions) => {
  const session = await getServerAuthSession();

  return {
    session,
    headers: opts.headers,
    prisma,
  };
};
```
So I want it everytime prisma client do any operation like find or update it will auto attach the tenant id

```javascript
const user = await prisma.user.findUnique({
  where: {
    email: 'emma@prisma.io',
    tenantId: session.tenantId //I want to omit this one and let a customize prisma client handle this
  },
  select: {
    email: true,
    posts: {
      select: {
        likes: true,
      },
    },
  },
})
```
```javascript
const user = await prisma.user.create({
  data: {
    email: 'elsa@prisma.io',
    name: 'Elsa Prisma',
    tenantId: session.tenantId //I want to omit this one and let a customize prisma client handle this
  },
})
```

**Reply by nurul3101** [reply, marked as answer]
Yes, that's possible with Client Extensions. Here is an [example](https://github.com/prisma/prisma-client-extensions/blob/main/input-transformation/script.ts) that demonstrated the same.

**Reply by nurul3101** [reply]
Thanks for letting us know that this discussion is resolved. I'll close this discussion. 🙏
